### PR TITLE
Get act working

### PR DESCRIPTION
**Pull request by raq929**
## Included Changes:

- Allows `act` to run the Lint JS and Ruby workflow locally
- Actually runs ruby tests

## Notes to Reviewer
Check that the output of running ruby tests locally is the same as the CI output.

### CODE REVIEW

File: .github/workflows/lint-and-test.yml [hunk 1]
```diff
@@ -27,9 +27,25 @@ jobs:
...
 
-      - name: Install Yarn
-        run: npm install -g yarn
+      # Remove when act supports services
+      - name: Install local dependencies
         if: ${{ env.ACT }}
+        run: |
+          npm install -g yarn
+          sudo apt-get update
+          sudo apt-get -yqq install libgconf-2-4 libatk1.0-0 libatk-bridge2.0-0 libgdk-pixbuf2.0-0 libgtk-3-0 libgbm-dev libnss3-dev libxss-dev libasound2 libxshmfence-dev postgresql postgresql-contrib
```
Comment (sequelster) [inline .github/workflows/lint-and-test.yml:36]: yikes. I know postgres requires a bunch of lib installs, but what the heck are 99% of these for?

File: .github/workflows/lint-and-test.yml [hunk 2]
```diff
@@ -27,9 +27,25 @@ jobs:
...
+      - name: Boot local dependencies
+        if: ${{ env.ACT }}
+        run: |
+          sudo service postgresql start
+
+      # Remove when act supports services
+      - name: Configure local dependencies
+        if: ${{ env.ACT }}
+        run: |
+          sudo -u postgres psql -U postgres -d postgres -c "alter user postgres with password 'postgres';"
```
Comment (sequelster) [inline .github/workflows/lint-and-test.yml:48]: aren't all of these the default for postgres??

File: .github/workflows/lint-and-test.yml [hunk 3]
```diff
@@ -64,7 +85,7 @@ jobs:
         run: bundle exec rails db:prepare
 
       - name: Ruby Tests
-        run: bundle exec rake test
+        run: bundle exec rake test spec
```
Comment (sequelster) [inline .github/workflows/lint-and-test.yml:88]: are you kidding me? You have to say "spec"??? THAT'S what we were missing??

**Review by sequelster — APPROVED** [review]
those act requirements are whack - I have questions. It's okay if you can't answer them. I'm saddened by what we were missing to run ruby tests. >_<

File: .github/workflows/lint-and-test.yml (hunk 1 above)
Comment (raq929) [reply .github/workflows/lint-and-test.yml:36]: Oh, I think I can take most of these out again, let me try it.

File: .github/workflows/lint-and-test.yml (hunk 3 above)
Comment (raq929) [reply .github/workflows/lint-and-test.yml:88]: YEP. Also `bundle exec rake test .` doesn't work, which _offends_ me.
This and the postgres connection stuff in `config/database.rb` were what was missing. 🙄

**Pull request merged**